subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -224,6 +226,31 @@ private constructor(
     inline operator fun <T : ModuleDependency> Configuration.invoke(dependency: T, dependencyConfiguration: T.() -> Unit): T =
         add(name, dependency, dependencyConfiguration)
 
+    /**
+     * Adds a dependency provider to the given configuration.
+     *
+     * @param dependency the dependency provider to be added.
+     * @param dependencyConfiguration the configuration to be applied to the dependency
+     *
+     * @see [DependencyHandler.addProvider]
+     * @since 6.9
+     */
+    @Incubating
+    operator fun Configuration.invoke(dependency: Provider<*>, dependencyConfiguration: ExternalModuleDependency.() -> Unit) =
```

Comment:
:question: Is there a reason this function takes a `Provider<*>` and the one below takes a `Provider<Any>`?

Reply:
Yes. I must say I'm no Kotlin expert but if it's `Provider<Any>` it just doesn't work in build scripts: the overload is not found and it thinks only the `String` version of the notation is supported...

Reply:
Could it be due to nullability inconsistencies in the API where the provider comes from?
`Provider<Any>` won't accept a `Provider<String?>`

Reply:
I don't know, but with `Provider<Any>`, the "candidates" are not `Provider<String>` vs `Provider<MinimalExternalDependency>`, it's `String` vs `Provider<MinimalExternalDependency>` and only the string version is reported by the compiler, as if the provider one wasn't fitting at all.